bin/unit-test-suite.mjs

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env node
+import util from 'util';
+import glob from 'glob';
+import runTask from './teamcity-task-runner.mjs';
+
+const globPromise = util.promisify( glob );
+
+function withTscInfo( { cmd, id } ) {
+	return {
+		testId: id,
+		name: 'yarn',
+		args: cmd,
+		env: { NODE_ENV: 'test' },
+	};
+}
+
+function withUnitTestInfo( cmd ) {
+	return {
+		testId: cmd.split( ' ' )[ 0 ],
+		name: 'yarn',
+		args: `${ cmd } --ci --reporters=default --reporters=jest-teamcity --silent`,
+	};
+}
+
+const allPackageTsconfigs = ( await globPromise( 'packages/*/tsconfig.json' ) ).join( ' ' );
+const tscPackages = withTscInfo( {
+	cmd: `tsc --build ${ allPackageTsconfigs }`,
+	id: 'type_check_packages',
+} );
+
+const tscCommands = [
+	{ cmd: 'tsc --noEmit --project apps/editing-toolkit/tsconfig.json', id: 'type_check_etk' },
+	{ cmd: 'tsc --noEmit --project client/tsconfig.json', id: 'type_check_client' },
+	{ cmd: 'tsc --noEmit --project test/e2e/tsconfig.json', id: 'type_check_tests' },
+].map( withTscInfo );
+
+// When Jest runs without --maxWorkers, each instance of Jest will try to use all
+// cores available. (Which is a lot in our CI.) This isn't a problem per se, because
+// everything ends up completing pretty quickly. However, with 100% CPU usage for
+// most of the test, it's possible for some tests which rely on i/o to time out.
+// This causes flakey tests. As a result, we need to manage the number of workers
+// manually so that there is some amount of margin.
+//
+// After some testing, I've found that 8+4 is a good setup. The largest task
+// runs by itself with 8 cores, and the other tasks run one by one with 4 other
+// cores. This leaves a final 4 cores free for tsc + any other tasks. This seems
+// to result in the fastest overall completion time.
+const testClient = withUnitTestInfo( 'test-client --maxWorkers=8' );
+const testPackages = withUnitTestInfo( 'test-packages --maxWorkers=4' );
+const testServer = withUnitTestInfo( 'test-server --maxWorkers=4' );
+const testBuildTools = withUnitTestInfo( 'test-build-tools --maxWorkers=4' );
```

Comment:
**comment** - I see how you pinned tasks to a specific number of cores. Useful to know!

Reply:
Yep! I think this is helpful when we want to run test suites with different complexity at the same time.